gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/DefaultKeystoreService.java

```diff
@@ -450,6 +481,22 @@ private void removeFromCache(String clusterName, String alias) {
 
   @Override
   public String getKeystorePath() {
-    return keyStoreDir + GATEWAY_KEYSTORE;
+    return config.getIdentityKeystorePath();
+  }
+
+  private char[] getIdentityKeystorePassword() throws KeystoreServiceException {
```

Comment:
`getIdentityKeystorePassword` and `getSigningKeystorePassword` has the same implementation except for the password alias they are fetching the credential for. What if you create a new private method `getKeystorePassword(String passwordAlias)` and you invoke this one with the proper params (`config.getIdentityKeystorePasswordAlias()` or `config.getSigningKeystorePasswordAlias()`)?

Reply:
Agreed... will change.